web3/src/main/java/org/hiero/mirror/web3/state/service/ContractStateService.java

```
@@ -0,0 +1,199 @@
+                    it.remove();
+                }
+                cachedSlotKeys.add(encodeSlotKey);
+                queriedSlotsCache.put(contractId, cachedSlotKeys);
+            }
+        } else {
+            cachedSlotKeys = new LinkedHashSet<>();
+            cachedSlotKeys.add(encodeSlotKey);
+            queriedSlotsCache.put(contractId, cachedSlotKeys);
+        }
+    }
+
+    /**
+     * Executes a batch query, returning slotKey-value pairs for contractId, then caches the result.
+     * The goal of the query is to preload previously requested data to avoid additional queries against the db.
+     * @param contractId id of the contract that the slotKey-value pairs are queried for.
+     * @param cachedSlotKeys slot keys used in the WHERE IN clause of the findStorageBatch query.
+     * @return slotKey-value pairs for contractId
+     */
+    private List<ContractSlotValue> findStorageBatch(final Long contractId, final Set<String> cachedSlotKeys) {
+        if (cachedSlotKeys != null && !cachedSlotKeys.isEmpty()) {
+            final var cachedSlots =
+                    cachedSlotKeys.stream().map(this::decodeSlotKey).toList();
+
+            final var existingSlotKeyValuePairs = contractStateRepository.findStorageBatch(contractId, cachedSlots);
+            final var existingSlots = existingSlotKeyValuePairs.stream()
+                    .map(pair -> ByteBuffer.wrap(pair.getSlot()))
+                    .collect(Collectors.toSet());
+            cacheSlotValues(contractId, existingSlotKeyValuePairs);
+
+            final var nonExistingSlots = cachedSlots.stream()
+                    .filter(key -> !existingSlots.contains(ByteBuffer.wrap(key)))
+                    .toList();
+            cacheNonExistingSlots(contractId, nonExistingSlots);
+
+            return existingSlotKeyValuePairs;
+        }
+        log.warn("findStorageBatch called with empty collection of slot keys for contractId: {}", contractId);
+        return new ArrayList<>();
```

We should not log a warning or allocate a list just because we have never called `findStorageBatch` for this contract id before. Remove this logic and invert logic to fail fast as is our standard.

```java
if (CollectionUtils.isEmpty(cachedSlotKeys)) {
    return List.of();
}
```

---

Removed the warning and inverted the if.